src/Microsoft.Fx.Portability/ObjectModel/DiagnosticAnalyzerInfo.cs

```diff
@@ -0,0 +1,23 @@
+﻿using System.Collections.Generic;
+using System.Text.RegularExpressions;
+using System;
+
+namespace Microsoft.Fx.Portability.ObjectModel
+{
+#if FEATURE_SERIALIZABLE
+    [Serializable]
+#endif
+    public class DiagnosticAnalyzerInfo
+    {
+        public string AnalyzerName { get; set; }
+        public string Id { get; set; }
+        public ICollection<CompatibilityRangeAttribute> Attributes { get; set; }
+        public bool IsCompatibilityDiagnostic { get; set; }
+
+        public int GetIdNumber()
+        {
+            var matches = Regex.Match(Id, @"(\d+)");
+            return int.Parse(matches.Captures[0].Value);
```

Comment:
What about if someone creates an entry like this

```csharp
var info = new DiagnosticAnalyzerInfo { Id == "NoNumber" };
var id = info.GetIdNumber();
```

I'm pretty sure it'll fail. You should check whether a capture group was found.